ecl/eclscheduler/eclscheduler.cpp

```diff
@@ -180,6 +181,258 @@ version: "1.0"
+
+        // stop and remove schedulers not in the new list as they are no longer needed
+        // for those in the new list, if they are already running, leave them alone
+        std::list<aindex_t> schedulersToBeRemoved;
+        ForEachItemIn(schedIdx, schedulers)
+        {
+            if (updatedQueuesSet.find(schedulers.item(schedIdx).getServerName().str()) == updatedQueuesSet.end())
+            {
+                // Queue has been removed
+                DBGLOG("Stopped listening to queue %s", schedulers.item(schedIdx).getServerName().str());
+                schedulers.item(schedIdx).stop();
+                schedulersToBeRemoved.emplace_front(schedIdx);
+            }
+            else
+            {
+                // "New" queue already running
+                updatedQueuesSet.erase(schedulers.item(schedIdx).getServerName().str());
+            }
+        }
+        for (auto &index : schedulersToBeRemoved)
+        {
+            schedulers.remove(index);
+        }
+
+        // only new schedulers will be left in the list
+        for (auto &qname : updatedQueuesSet)
+        {
+#ifdef _CONTAINERIZED
+            DBGLOG("Start listening to queue %s", qname.c_str());
+            Owned<EclScheduler> scheduler = new EclScheduler(this, qname.c_str());
+            scheduler->start();
+            schedulers.append(*scheduler.getClear());
+#else
+            Owned<IStringIterator> targetClusters = getTargetClusters("EclSchedulerProcess", qname.c_str());
+            ForEach(*targetClusters)
+            {
+                SCMStringBuffer targetCluster;
+                targetClusters->str(targetCluster);
+                DBGLOG("Start listening to queue %s", targetCluster.str());
+                Owned<EclScheduler> scheduler = new EclScheduler(this, targetCluster.str());
+                scheduler->start();
+                schedulers.append(*scheduler.getClear());
+            }
+#endif
+        }
+    }
+
+    void stop()
+    {
+        DBGLOG("stop()");
+        CriticalBlock b(updateSchedulersCS);
+        ForEachItemIn(schedIdx, schedulers)
+        {
+            schedulers.item(schedIdx).stop();
+        }
+    }
+
+    std::unordered_set<std::string> splitBy(const char *stringList, const char *delim)
+    {
+        std::unordered_set<std::string> updatedQueuesList;
+        if (!stringList || !*stringList || !delim || !*delim)
+            return updatedQueuesList;
+
+        std::string str(stringList);
+        std::regex del(delim);
+        std::sregex_token_iterator it(str.begin(), str.end(), del, -1);
+        std::sregex_token_iterator end;
+        while (it != end)
+        {
+            updatedQueuesList.emplace(*it);
+            ++it;
+        }
+        return updatedQueuesList;
+    }
+
+    CConfigUpdateHook updateConfigHook;
+
+    CIArrayOf<EclScheduler> schedulers;
+    CriticalSection updateSchedulersCS;
+    StringAttr currentQueueNames;
+    StringAttr updatedQueueNames;
```

Comment:
see other comments, think can consolidate on a single 'scheduerQueues' that updateSchedulers() references against the names in 'schedulers'.